### ISSUE TITLE
Logic Errors, Expiry of SESS session

### ISSUE THREAD

**Issue by KAZIcodes** [mohsen-mirzaei/available-class-notifier#1]
Good job dude, but it got below bugs:

Logic Errors:
1. All sections are iterated for all subjects
2. When a free capacity is found for a subject, it will go into an infinite loop leading to not checking other subjects and sections

Expiry of SESS session:
SESS expires user's session after a couple of minutes, so you should consider handling that too; You probably also need to bypass captcha too :)

Hope this issue helps ;)

**Comment by mohsen-mirzaei** [comment]
Thanks for your feedback.

Session expiry:
Each session lasts a few minutes when the user is inactive and in this case since the user is expected to want to check availability pretty frequently, there shouldn't be much of an issue with session expiry. Granted it would be nice to be able to have longer wait times between checks, however, in my opinion it wouldn't be worth the hassle to bypass CAPTCHA.

Not checking after free capacity is found:
Since the user is expected to want to enter their SESS account when free capacity is found and considering that SESS only allows one simultaneous session, it is not possible to continue to check other classes.

Iterating the same sections for each class:
This project was mostly a way for me to take the "danesh khanevade" class :) and for now, is not tested for more than one subject. A few small changes including the one you mentioned are needed to support multiple subjects. Will apply them soon. hopefully.

**Comment by KAZIcodes** [comment]
I agree with your reasoning about session expiry part but about the second part, well you are correct but this limitation is because of the way that the script is coded which is opening an instance of chrome and ... , instead if you work with "requests" and "BeautifulSoup" libraries you can code the script in a way that after opening the chrome instance and user login, it takes the session and use those libraries to check for availability of the courses the user wants and notify the user for availability and then the user should only use the same opened chrome instance to pick up the available course because of the single simultaneous session; this way the script can continue running even when the user is working with SESS using the chrome instance; But to implement this method you should analyze HTTP requests of SESS app and maybe handle some details among it's functionalities too.

A really better way to code it though is to let the script do all the things, meaning Login, Identifying available courses and Picking them up for the user too which can be done using the same mentioned libraries but the disadvantage of this scenario is that the user can not sign in and work with SESS in a browser when the script is running because of the  single simultaneous session thing (although you can open a chrome instance for the user with the logged in session that the script is working with so that if a user needed to work with SESS while the script is running it would be possible) and remember if you want to make the script perfect you should implement this scenario multi-threaded but I think that wouldn't be necessary for now :) 

Nevertheless, the script works fine for a single course and these are only suggestions to make it better if you are interested ;)
